gst/gst_allocator.go

```diff
@@ -111,3 +111,43 @@ func (a *Allocator) Alloc(size int64, params *AllocationParams) *Memory {
 func (a *Allocator) Free(mem *Memory) {
 	C.gst_allocator_free(a.Instance(), mem.Instance())
 }
+
+// DmaBufAllocator is a go representation of a GstDmaBufAllocator
+type DmaBufAllocator Allocator
+
+// DmaBufAllocator returns a new GstDmaBufAllocator.
+func NewDmaBufAllocator() *DmaBufAllocator {
+	return (*DmaBufAllocator)(wrapAllocator(glib.TransferFull(unsafe.Pointer(C.gst_dmabuf_allocator_new()))))
+}
+
+// Instance returns the underlying GstAllocator instance.
+func (a *DmaBufAllocator) Instance() *C.GstAllocator { return C.toGstAllocator(a.Unsafe()) }
+
+// AllocDmaBuf is used to allocate a new memory block with memory that is at least size big.
+func (a *DmaBufAllocator) AllocDmaBuf(fd int, size int64) *Memory {
+	mem := C.gst_dmabuf_allocator_alloc(a.Instance(), C.gint(fd), C.gsize(size))
+	return FromGstMemoryUnsafeFull(unsafe.Pointer(mem))
+}
+
+// AllocDmaBufWithFlags is used to allocate a new memory block with memory that is at least size big.
+func (a *DmaBufAllocator) AllocDmaBufWithFlags(fd int, size int64, flags FdMemoryFlags) *Memory {
+	mem := C.gst_dmabuf_allocator_alloc_with_flags(a.Instance(), C.gint(fd), C.gsize(size), C.GstFdMemoryFlags(flags))
+	return FromGstMemoryUnsafeFull(unsafe.Pointer(mem))
+}
+
+// FdAllocator is a go representation of a GstFdAllocator
+type FdAllocator Allocator
```

Comment:
Doesn't this hide the methods of the base allocator? I think the following would be more useful:

```go
type FdAllocator struct {
    *gst.Allocator
}
```

Reply:
That was intended: the only method of the base allocator that would be useful here is `Instance()`, `Alloc` doesn't make much sense for them as far as I understand.

Reply:
I don't know the allocators well enough for this. @sdroege can you help please?